File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/HouseholdDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record HouseholdDto(
    UUID id,
    String name,
    Instant createdAt,
    List<HouseholdMemberDto> members
) {
    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public List<HouseholdMemberDto> getMembers() {
        return members;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/HouseholdDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.time.Instant
import java.util.UUID

data class HouseholdDto(
    val id: UUID,
    val name: String,
    val createdAt: Instant,
    val members: List<HouseholdMemberDto>,
)
